testing/python/fixture.py

```diff
@@ -3977,3 +3978,71 @@ def test_func(self, f2, f1, m2):
         items, _ = testdir.inline_genitems()
         request = FixtureRequest(items[0])
         assert request.fixturenames == "s1 p1 m1 m2 c1 f2 f1".split()
+
+    def test_multiple_packages(self, testdir):
+        """Complex test involving multiple package fixtures. Make sure teardowns
+        are executed in order.
+        .
+        └── root
+            ├── __init__.py
+            ├── sub1
+            │   ├── __init__.py
+            │   ├── conftest.py
+            │   └── test_1.py
+            └── sub2
+                ├── __init__.py
+                ├── conftest.py
+                └── test_2.py
+        """
+        root = testdir.mkdir("root")
+        root.join("__init__.py").write("values = []")
+        sub1 = root.mkdir("sub1")
+        sub1.ensure("__init__.py")
+        sub1.join("conftest.py").write(
+            textwrap.dedent(
+                """\
+            import pytest
+            from .. import values
+            @pytest.fixture(scope="package")
+            def fix():
+                values.append("pre-sub1")
+                yield values
+                values.pop()
+        """
+            )
+        )
+        sub1.join("test_1.py").write(
+            textwrap.dedent(
+                """\
+            from .. import values
+            def test_1(fix):
+                assert values == ["pre-sub1"]
+        """
+            )
+        )
+        sub2 = root.mkdir("sub2")
+        sub2.ensure("__init__.py")
+        sub2.join("conftest.py").write(
+            textwrap.dedent(
+                """\
```

Comment:
To avoid this weird line break by black I usually put the triple strings into a variable:

```python
contents = dedent("""\
    ...
""")
sub2.join("conftest.py").write(contents)
```

But really up to you, just in case this bothers you. 😉

Reply:
black is going to reformat that to:

```python
contents = dedent(
    """\
    ...
"""
    )
```

:rip:

Reply:
Oh. 😞

Reply:
you can do this:

```python
x = """\
    ...
"""
print(dedent(x))
```
but then. meh.